test/validation/api/ipsec/ipsec_test_out.c

```diff
@@ -375,6 +407,10 @@ static void test_esp_out_in_all(void)
 	for (c = 0; c < ARRAY_SIZE(ciphers); c++)
 		for (a = 0; a < ARRAY_SIZE(auths); a++)
 			test_esp_out_in(&ciphers[c], &auths[a]);
+
+	for (c = 0; c < ARRAY_SIZE(cipher_auth_comb); c++)
+		test_esp_out_in(&cipher_auth_comb[c].cipher,
```

Comment:
AFAICS, this does not add any new tests but duplicates existing tests that are currently separate test cases and not part of test_esp_out_in_all().

This change might have some merit as a code cleanup patch if it also removed the existing test cases.

Reply:
I saw that there were tests added to make sure all tests use a common framework. Thought I can chip in by extending it to include combined mode algos as well. I can remove the existing cases when we decide to remove the ones covered with the first loop (non-combined mode cipher & auth).

I saw that earlier every combination was tried as a separate test case, while with this approach, the skipped tests are not really counted. So I wasn't sure if this is still WIP. IMO, IPsec tests can focus on other IPsec features rather on combinations of algos, and the direction looks sensible to me.

Reply:
> I can remove the existing cases when we decide to remove the ones covered with the first loop (non-combined mode cipher & auth).

I do not think there are any existing separate test cases that are also covered buy the common out+in test.

A while ago I changed the code to get rid of duplication and to be able to add more algorithm combos without combinatoric explosion of the number of separate test cases. I am not working on further refactoring, but some may be needed when further tests are added (e.g. if we start supporting more than one ICV length per algorithm).